app/models/minion.rb

```diff
@@ -18,6 +18,8 @@ def assign_role(role:)
       salt.assign_role role: role
     end
     true
+  rescue
```

Comment:
I suggest we rescue explicitly only what makes sense. For example I would rollback the transaction but I would reraise in the case of ```Errno::ECONNREFUSED``` because that probably means something is wrong.